src/main/java/appeng/util/Platform.java

```diff
@@ -1281,56 +1280,36 @@ else if( entity != null && pos == null )
 		return input;
 	}
 
+	@SuppressWarnings( { "rawtypes", "unchecked" } )
 	public static void postChanges( final IStorageGrid gs, final ItemStack removed, final ItemStack added, final IActionSource src )
 	{
-
-		final IItemStorageChannel itemChannel = AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class );
-		final IFluidStorageChannel fluidChannel = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class );
-		final IItemList<IAEItemStack> itemChanges = itemChannel.createList();
-		final IItemList<IAEFluidStack> fluidChanges = fluidChannel.createList();
-
-		if( !removed.isEmpty() )
+		for( final IStorageChannel<?> chan : AEApi.instance().storage().storageChannels() )
 		{
-			final IMEInventory<IAEItemStack> myItems = AEApi.instance().registries().cell().getCellInventory( removed, null, itemChannel );
+			final IItemList<?> myChanges = chan.createList();
 
-			if( myItems != null )
+			if( !removed.isEmpty() )
 			{
-				for( final IAEItemStack is : myItems.getAvailableItems( itemChanges ) )
+				final IMEInventory myInv = AEApi.instance().registries().cell().getCellInventory( removed, null, chan );
+				if( myInv != null )
 				{
-					is.setStackSize( -is.getStackSize() );
+					myInv.getAvailableItems( myChanges );
+					for( final IAEStack is : myChanges )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)